wgpu-core/src/error.rs

```diff
@@ -0,0 +1,321 @@
+use core::fmt;
+use std::error::Error;
+
+use crate::{
+    gfx_select,
+    hub::{Global, IdentityManagerFactory},
+};
+
+pub trait AsDisplay {
```

Comment:
Looks like this is only needed for just a few cases. Could it be worked around?
Something like https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ced1f8a2ddc2c0b986421b1acd195588

Reply:
Yeah, I don't remember the whole thought process why it originally was like that, but that looks like workable solution.

Reply:
Hmm, couldn't get it quite to work.
```
pub fn format_pretty_any(
    global: &Global<IdentityManagerFactory>,
    error: &(dyn Error + 'static),
) -> String
```
In here, even though `Error´ requires Display, somehow it won't allow casting to it. Adding the trait to the trait object requirement doesn't work and suggests making a wrapper trait that requires the two things, but `Error` already requires `Display`, so that seems like a dead-end.

```
pub trait PrettyError: Error {
    fn fmt_pretty(&self, _global: &Global<IdentityManagerFactory>) -> String {
        let self_display = self as &dyn fmt::Display;
        format_error_line(self_display)
    }
}
```
In here it says self doesn't have compile time size, though I'm not entirely sure why it requires that, isn't trait object a compile time known size?

Either I'm missing some syntax that I'm forgetting, or its not really possible to do it like that.